pkg/oidcclient/login.go

```diff
@@ -260,6 +261,7 @@ func (h *handlerState) baseLogin() (*oidctypes.Token, error) {
 	// If the ID token is still valid for a bit, return it immediately and skip the rest of the flow.
 	cached := h.cache.GetToken(cacheKey)
 	if cached != nil && cached.IDToken != nil && time.Until(cached.IDToken.Expiry.Time) > minIDTokenValidity {
+		plog.Debug("Pinniped: Found unexpired cached token")
```

Comment:
Since this is part of our public Go-level API, I think we should avoid logging to a global logger and pass in a `logr.Logger` (or maybe even a simpler `debugHook func(msg string)` handler) instead.

This would allow someone consuming the library to fully control the user interface (e.g., they could send debug output to a UI component if they were building a non-terminal app).

Maybe something kind of like the "error handler" hook in the session cache package? https://github.com/vmware-tanzu/pinniped/blob/620a4d55b7c6c15d50888e76f03f443888243719/pkg/oidcclient/filesession/filesession.go#L35-L39

Then when we instantiate this login handler thing from the CLI code, we can pass in the existing `pLogger` instance you created there and we'd get the prefixing automatically as well.

Reply:
Okay, something like adding a logger to the `handlerState` and creating a `WithLogger` function?
I was thinking through what might be a useful default value-- do you think the global logger is okay in that case as long as you have the option to specify a different one? Or we could instantiate a new one inside `Login`.

Reply:
Yeah, that sounds good. I think the best default would be a no-op logger that just discards the message.

Reply:
Ok, made the change to use a `logr.Logger` with a default of `DiscardLogger`.